### PR TITLE
fix: support android app

### PR DESCRIPTION
**Pull request by oldkingOK**
See [#52](https://github.com/artisticat1/obsidian-latex-suite/issues/52#issuecomment-1722239752)

### CODE REVIEW

**Comment by artisticat1** [comment]
Thanks for the PR! Is this ready for review, or does it still need further testing and debugging?

**Comment by oldkingOK** [comment]
Yeah, it still needs further testing and debugging.

**Comment by oldkingOK** [comment]
I'm sorry for the misoperation, I'm not so familiar with it because it's my first time opening a PR.
Btw it is ready for review.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# Obsidian Latex Suite <img src="https://img.shields.io/github/manifest-json/v/artisticat1/obsidian-latex-suite"> <img src="https://img.shields.io/github/downloads/artisticat1/obsidian-latex-suite/total">
+# Obsidian Latex Suite `<img src="https://img.shields.io/github/manifest-json/v/artisticat1/obsidian-latex-suite">` `<img src="https://img.shields.io/github/downloads/artisticat1/obsidian-latex-suite/total">`
```
Comment (artisticat1) [inline README.md:1]: It looks like the formatting in the readme has gotten muddled up. Could you revert the changes to the formatting?

File: src/latex_suite.ts [hunk 2]
```diff
@@ -109,7 +116,8 @@ export const handleTabstops = (view: EditorView) =>
 // CodeMirror extensions that are required for Latex Suite to run
 export const latexSuiteExtensions = (settings: LatexSuiteProcessedSettings) => [
 	getLatexSuiteConfigExtension(settings),
-	Prec.highest(EditorView.domEventHandlers({"keydown": onKeydown})), // Register keymaps
+	Prec.highest(EditorView.domEventHandlers({"keydown": 
+		Platform.isAndroidApp ? onAndroidKeydown : onKeydown})), // Register keymaps. Compatibility for android users with IME keyboard
```
Comment (artisticat1) [inline src/latex_suite.ts:120]: Is it possible to remove the `Platform.isAndroidApp` check and make the fix platform-independent? The goal is to make Latex Suite a pure CodeMirror extension, with no dependency on Obsidian. This will allow it to be embedded in any other CM editor, e.g. on the web.

File: README.md (hunk 1 above)
Comment (oldkingOK) [reply README.md]: Sorry for the late reply. A vscode extension I was using reformated the readme and I have restored it.

File: src/latex_suite.ts (hunk 2 above)
Comment (oldkingOK) [reply src/latex_suite.ts]: Ok, then I need to edit some code in latex_suite.ts to make this fix platform-independent, which requires more testing and debugging.

**Comment by oldkingOK** [comment]
Hi @IEatCodeDaily ,@awsharif,
Can you guys help me test the [latest build](https://github.com/oldkingOK/obsidian-latex-suite/releases/tag/v0.0.4) , I edited some key parts of the code and need a lot of testing work, thanks!

**Comment by IEatCodeDaily** [comment]
Sorry, didn't check email for quite a while. Didn't know I got tagged here.

Just tested it on Samsung A50 and Samsung Galaxy Tab S6 (w/ and w/out physical keyboard).
Work flawlessly with Samsung Keyboard, Gboard, MS SwiftKey, and Simple Keyboard all with auto-correction enabled.
Auto-correction still works on Samsung Keyboard and Gboard, but doesn't work on MS SwiftKey and Simple Keyboard.
Not a big deal when typing using physical keyboard, but a minor inconvenience when using on-screen keyboard.

The same double snippets entry bug I mentioned in #52 still happen when make.md is also installed though. But I have no idea whether the fault is on make.md or latex-suite.

**Comment by DuncanBH** [comment]
Heya I just set up the plugin (V1.9.0) on my Tab S9 (Android 14, One UI 6.0), I've run into similar issues as to what's been described here and in https://github.com/artisticat1/obsidian-latex-suite/issues/213
Haven't had any luck with the on screen keyboard, I did get it working with a physical keyboard by turning off the predictive text. It was a Bluetooth keyboard, I'll test out it on the official flipbook keyboard case when I get my hands on one soon.

Is there another solution I've missed somewhere? I'd like the predictive text but I suppose I could just switch it off when using the plugin.

**Comment by oldkingOK** [comment]
> @oldkingOK Hello,The link https://github.com/oldkingOK/obsidian-latex-suite/releases/tag/v0.0.4 is not working. Could you please re-upload it? Thank you very much.

@CamWam 
Sorry for the delay. This link leads to one of my previous and outdated repositories. I recently updated the code, can you test this below?
https://github.com/oldkingOK/obsidian-latex-suite/releases/tag/1.9.7-dev
